### PR TITLE
Improve `datafusion-cli` print format tests

### PR DESCRIPTION
**Pull request by alamb**
Draft as it build on https://github.com/apache/arrow-datafusion/pull/8895 

## Which issue does this PR close?

Related to https://github.com/apache/arrow-datafusion/issues/8702

Follow on to https://github.com/apache/arrow-datafusion/pull/8895 

## Rationale for this change

While working on https://github.com/apache/arrow-datafusion/pull/8895 I found the tests somewhat hard to follow and it was hard for me to understand if there were existing tests that covered I was changing.

It was also hard top understand the coverage in general given how the tests were structured

## What changes are included in this PR?
1. Rework tests in print_format to be more explict
2. Add additional coverage for TSV and empty result printing

## Are these changes tested?
This PR is all tests (no code changes)

## Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion-cli/src/print_format.rs [hunk 1]
```diff
@@ -161,152 +161,263 @@ impl PrintFormat {
...
+        for format in [
+            PrintFormat::Csv,
+            PrintFormat::Tsv,
+            PrintFormat::Table,
+            PrintFormat::Json,
+            PrintFormat::NdJson,
+            PrintFormat::Automatic,
+        ] {
+            // no output for empty batches, even with header set
+            PrintBatchesTest::new()
```
Comment (alamb) [inline datafusion-cli/src/print_format.rs:210]: I think this style of test is clearer what is expected and what is covered

File: datafusion-cli/src/print_format.rs [hunk 2]
```diff
@@ -161,152 +161,263 @@ impl PrintFormat {
...
+            "| 1 | 4 | 7 |",
+            "| 2 | 5 | 8 |",
+            "| 3 | 6 | 9 |",
+            "+---+---+---+",
+        ];
+
+        PrintBatchesTest::new()
+            .with_format(PrintFormat::Table)
+            .with_batches(split_batch(three_column_batch()))
+            .with_header(WithHeader::Ignored)
```
Comment (alamb) [inline datafusion-cli/src/print_format.rs:303]: Now there is additional coverage for formats that ignore the `with_header` flag

**Review by comphead — APPROVED** [review]
lgtm thanks @alamb

**Review by Dandandan — APPROVED** [review]
